crates/polars-io/src/partition.rs

```diff
@@ -127,3 +128,107 @@ where
     }
     path
 }
+
+pub fn write_partitioned_dataset<S>(
```

Comment:
Created `write_partitioned_dataset` here in `polars-io`.

I was considering putting a `fn write_parquet_partitioned` into `impl DataFrame`, but I notice that on the rust side we don't have e.g. `DataFrame::write_parquet` and others, so I just made it a function like this